addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationUpnpServer.java

```diff
@@ -105,18 +108,24 @@ public void run() {
                         logger.trace("Got SSDP Discovery packet from {}:{}", recv.getAddress().getHostAddress(),
                                 recv.getPort());
                         if (data.startsWith("M-SEARCH")) {
-                            String msg = String
-                                    .format(discoString,
-                                            "http://" + address.getHostAddress().toString() + ":"
-                                                    + System.getProperty("org.osgi.service.http.port") + discoPath,
-                                            usn);
-                            DatagramPacket response = new DatagramPacket(msg.getBytes(), msg.length(),
-                                    recv.getAddress(), recv.getPort());
-                            try {
-                                logger.trace("Sending to {} : {}", recv.getAddress().getHostAddress(), msg);
-                                sendSocket.send(response);
-                            } catch (IOException e) {
-                                logger.error("Could not send UPNP response", e);
+                            String hueId = usn.substring(usn.length() - 12).toUpperCase();
+
+                            String[] stVersions = { "upnp:rootdevice", "urn:schemas-upnp-org:device:basic:1",
+                                    "uuid:" + usn };
+
+                            for (String st : stVersions) {
+                                String msg = String.format(discoString, MULTI_ADDR, UPNP_PORT_RECV,
+                                        "http://" + address.getHostAddress().toString() + ":" + webPort + discoPath,
+                                        hueId, st, usn);
+
+                                DatagramPacket response = new DatagramPacket(msg.getBytes(), msg.length(),
+                                        recv.getAddress(), recv.getPort());
+                                try {
+                                    logger.trace("Sending to {} : {}", recv.getAddress().getHostAddress(), msg);
+                                    sendSocket.send(response);
+                                } catch (IOException e) {
+                                    logger.error("Could not send UPNP response", e);
```

Comment:
@kaikreuzer I believe IOException are likely to happen and should not be reported at ERROR level or do I still not get the/your logic 😉